### PR TITLE
feat(if-else): add if-else access control

### LINKED ISSUE #40: feat: if-else
Show an example of control flow with if-else.
This can be a pure cairo program, not necessarily a contract.

### PR DESCRIPTION
**Pull request by HansBhatia**
Closes #40

### CODE REVIEW

File: listings/ch00-introduction/if_else/src/access_control.cairo [hunk 1]
```diff
@@ -0,0 +1,23 @@
+use starknet::ContractAddress;
+
+#[contract]
+mod SimpleAccessControl {
+    use starknet::get_caller_address;
+
+    struct Storage {
+        owner: starknet::ContractAddress,
```
Comment (enitrat) [inline listings/ch00-introduction/if_else/src/access_control.cairo:8]: by convention, it's better to import it in the contract with `use starknet::{get_caller_address,ContractAddress}` and just use `ContractAddress`

File: listings/ch00-introduction/if_else/src/access_control.cairo [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        owner: starknet::ContractAddress, 
+    }
+
+    #[constructor]
+    fn constructor(_address: starknet::ContractAddress) {
+        owner::write(_address);
+    }
+
+    #[external]
+    fn onlyOwner() -> bool {
```
Comment (enitrat) [inline listings/ch00-introduction/if_else/src/access_control.cairo:17]: `snake_case`

File: listings/ch00-introduction/if_else/src/access_control.cairo [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        if (get_caller_address() == owner::read()) {
```
Comment (enitrat) [inline listings/ch00-introduction/if_else/src/access_control.cairo:18]: in that case, we can just simplify it to `return get_caller_address() == owner::read()`, it's more idiomatic 😄

File: listings/ch00-introduction/if_else/src/access_control.cairo [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    struct Storage {
+        owner: starknet::ContractAddress, 
+    }
+
+    #[constructor]
+    fn constructor(_address: starknet::ContractAddress) {
+        owner::write(_address);
+    }
+
+    #[external]
```
Comment (enitrat) [inline listings/ch00-introduction/if_else/src/access_control.cairo:16]: this should be a `#[view]` function, given that there is no associated state change

**Comment by enitrat** [comment]
> added if_else #40

A good practice is to link your PRs to existing issues with predefined keywords, like `Closes`: e.g. 
Closes #40 

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests

File: listings/ch00-introduction/if_else/src/access_control.cairo [hunk 5]
```diff
@@ -1,23 +1,30 @@
...
-        owner: starknet::ContractAddress, 
+        owner: ContractAddress, 
     }
 
+    #[event]
+    fn welcomeEvent(name: felt252) {}
+
     #[constructor]
-    fn constructor(_address: starknet::ContractAddress) {
+    fn constructor(_address: ContractAddress) {
```
Comment (enitrat) [inline listings/ch00-introduction/if_else/src/access_control.cairo:13]: Just one last remark on convention: For now, the common pattern is to have parameters without `_` prefix, and storage variables with a `_` prefix.

Also, events are written in `PascalCase`

File: listings/ch00-introduction/if_else/src/access_control.cairo [hunk 6]
```diff
@@ -1,23 +1,30 @@
...
         owner::write(_address);
     }
 
+    #[internal]
```
Comment (enitrat) [inline listings/ch00-introduction/if_else/src/access_control.cairo:17]: In that case, the `#[internal]` keyword is not required - everything that is neither `#[external]` nor `#[view]` is internal by default.

**Review by enitrat — APPROVED** [review]
Ready to merge after addressing the last small comments ;)

**Comment by HansBhatia** [comment]
Changes made!

**Comment by enitrat** [comment]
I just realized @HansBhatia, you forgot to add a `markdown` page with a short explanation of `if-else` and the code written

**Comment by HansBhatia** [comment]
@enitrat I believe the changes are made!

**Comment by enitrat** [comment]
Great! 
You also need to add the chapter to the `SUMMARY.md` file for it to appear in the book.
Also, I renamed "control" to "log", given that our contract is no longer controlling anything, but just logging.

**Pull request merged**